### PR TITLE
Add GKE node pool e2e test. Fix GKE local SSD test to only run under GKE.

### PR DESCRIPTION
**Pull request by fabioy**
Just adding a node pool e2e test for GKE runs. And minor fix to local SSD test to only run under GKE.

[Analytics]()

### CODE REVIEW

File: test/e2e/gke_node_pools.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	framework.Logf("Checking that no nodes have the deleted node pool's label.")
+	nodeCount = nodesWithPoolLabel(f, poolName)
+	if nodeCount != 0 {
+		framework.Failf("Wanted 0 nodes with node pool label, got: %v", nodeCount)
+	}
+	framework.Logf("Success, found no nodes with the deleted node pool's label.")
+}
+
+// nodesWithPoolLabel returnw the number of nodes that have the "gke-nodepool"
```
Comment (roberthbailey) [inline test/e2e/gke_node_pools.go]: returns

**Comment by roberthbailey** [comment]
lgtm modulo a nit

**Comment by fabioy** [comment]
Fixed typo. Adding LGTM label. Thanks.

**Comment by fabioy** [comment]
@k8s-bot: test this issue: #22655

**Comment by fabioy** [comment]
@k8s-bot: test this issue: #22655

**Comment by fabioy** [comment]
@k8s-bot: test this issue: #22655

**Comment by fabioy** [comment]
@k8s-bot e2e test this issue: #26431

**Comment by roberthbailey** [comment]
@spxtr -- any idea what "Jenkins verification — Build finished." means?

**Comment by spxtr** [comment]
That was @mikedanese's experiment to split the unit/integration job into two. It didn't go smoothly, and I think you can ignore the X. I don't thing the submit queue looks at it.

**Comment by fabioy** [comment]
@k8s-bot: test this issue: #IGNORE

**Comment by goltermann** [comment]
Still have an X next to Jenkins verification and this isn't the queue

@k8s-bot: test this issue: #IGNORE

**Comment by spxtr** [comment]
@k8s-bot verify this issue: #IGNORE

**Pull request merged**